File: csharp/SearchSort.cs
using System;

public static class SearchSort
{
    public static bool LinearSearch(int[] array, int target)
    {
        foreach (int item in array)
        {
            if (item == target) return true;
        }
        return false;
    }

    public static void BubbleSort(int[] array)
    {
        int n = array.Length;
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n - i - 1; j++)
            {
                if (array[j] > array[j + 1])
                {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }

    public static void RunSearchSortDemo()
    {
        Console.WriteLine("\n🔍 Linear Search Demo");
        int[] values = { 3, 8, 1, 4, 9 };
        Console.WriteLine("Values: " + string.Join(", ", values));
        Console.WriteLine("Searching for 4: " + LinearSearch(values, 4));
        Console.WriteLine("Searching for 10: " + LinearSearch(values, 10));

        Console.WriteLine("\n🌀 Bubble Sort Demo");
        int[] unsorted = { 5, 2, 9, 1, 7 };
        Console.WriteLine("Before sort: " + string.Join(", ", unsorted));
        BubbleSort(unsorted);
        Console.WriteLine("After sort:  " + string.Join(", ", unsorted));
    }
}

File: python/search_sort.py
def linear_search(array, target):
    for item in array:
        if item == target:
            return True
    return False


def bubble_sort(array):
    n = len(array)
    for i in range(n):
        for j in range(n - i - 1):
            if array[j] > array[j + 1]:
                array[j], array[j + 1] = array[j + 1], array[j]


def run_search_sort_demo():
    print("\n🔍 Linear Search Demo")
    values = [3, 8, 1, 4, 9]
    print("Values:", values)
    print("Searching for 4:", linear_search(values, 4))  # True
    print("Searching for 10:", linear_search(values, 10))  # False

    print("\n🌀 Bubble Sort Demo")
    unsorted = [5, 2, 9, 1, 7]
    print("Before sort:", unsorted)
    bubble_sort(unsorted)
    print("After sort: ", unsorted)
